### DISCUSSION TITLE
Running actix-web hello world sample as a windows service

### DISCUSSION THREAD

**Discussion by nikhilramabhadra** [actix/actix-web#2680] [Q&A]
How do I run the actix-web [hello world sample](https://docs.rs/actix-web/latest/actix_web/#examples) as a windows service? Has anyone done this? Sample code?

[Windows Service](https://github.com/mullvad/windows-service-rs) - This is a crate for windows service but the actix-web sample has an async main.

**Comment by kennethuil** [comment]
https://docs.rs/actix-web/latest/actix_web/rt/index.html - the Windows Service should explicitly spin up an Actix runtime and block on it.  That's basically what "async main" does anyway.

**Reply by nikhilramabhadra** [reply]
I had a go at writing code for this. Attached the file. I used the Service Control Manager sc command to create the service and then tried to start it in services.msc but the UI timed out.
[main.txt](https://github.com/actix/actix-web/files/8187536/main.txt)

Will check with [windows-service-rs](https://github.com/mullvad/windows-service-rs)

**Reply by nikhilramabhadra** [reply]
Got the service working using sample code from [windows-service-rs](https://github.com/mullvad/windows-service-rs).

The issue now is that I can't get the service to stop. I see the following error:

_**Windows could not stop the actix_service2 service on Local Computer.
The service did not return an error.  This could be an internal Windows error or an internal service error.
If the problem persists, contact your system administrator.**_

[main.txt](https://github.com/actix/actix-web/files/8199502/main.txt)
[Cargo_toml.txt](https://github.com/actix/actix-web/files/8199503/Cargo_toml.txt)
